### DISCUSSION TITLE
How to pass authentication tokens cookie to `getClientEnvironment()` function

### DISCUSSION THREAD

**Discussion by sahidrahman404** [RevereCRE/relay-nextjs#96] [Q&A]
Hi. What is the best way for passing an authentication token stored in a cookie to the getClientEnvironment() function? The resource I'm querying requires an authorization bearer token

**Comment by rrdelaney** [comment]
Hey @sahidrahman404, it's a bit hard to give you a better answer due to lack of context, but I would generally recommend grabbing the cookie using the [`document.cookie` API](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie) inside of your network fetch function.

**Reply by sahidrahman404** [reply]
The issue I encountered is that I couldn't access the HTTP-only cookie using the `Document.cookie` property. I ended up creating a helper API that retrieves the cookie session and passes it to the client, like the below
```ts
// Create a network request to retrieve the cookie session for GraphQL requests
export function createClientNetwork() {
  return Network.create(async (params, variables) => {
    // Fetch the token from the API endpoint
    const res = await fetch("http://localhost:3000/api/tokens/get");
    const token = await res.json() as string | null;

    // Make a GraphQL request with the retrieved token
    const json = await makeGraphQLRequest({
      token: token,
      text: params.text,
      variables: variables
    });

    return json;
  });
}
```
